osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -168,9 +168,12 @@ private void beatmapSetsChanged(object? beatmaps, NotifyCollectionChangedEventAr
                         int previousIndex = Items.IndexOf(beatmap);
                         Debug.Assert(previousIndex >= 0);
 
+                        // we're intentionally being lenient with there being two difficulties with equal online ID or difficulty name.
+                        // this can be the case when the user modifies the beatmap using the editor's "external edit" feature.
+                        // TODO: this should probably be fixed somewhere, this doesn't make sense as it is.
                         BeatmapInfo? matchingNewBeatmap =
-                            newSetBeatmaps.SingleOrDefault(b => b.OnlineID > 0 && b.OnlineID == beatmap.OnlineID) ??
-                            newSetBeatmaps.SingleOrDefault(b => b.DifficultyName == beatmap.DifficultyName && b.Ruleset.Equals(beatmap.Ruleset));
+                            newSetBeatmaps.FirstOrDefault(b => b.OnlineID > 0 && b.OnlineID == beatmap.OnlineID) ??
```

Comment:
Can we leave the stringent condition on online id if the test case is the only scenario which can actually reproduce?

Having two beatmaps with the same online ID in the local database sounds like an impossible case.

Reply:
@bdach thoughts on this? we really should not be able to get in a state where two online IDs exist in the database that conflict, correct?

Reply:
Also cannot repro using above instructions. The duplicate beatmap does seem to retain the duplicate online ID in its file, but doesn't crash song select (I can't open my realm database right now due to outdated realm studio so can't check there 😦)

Reply:
Turns out the repro scenario is a bit convoluted and as specific as it was in the issue thread. Select a difficulty, mount for external edit, clone the selected difficulty, change the name of the selected difficulty to something else, keep the clone with the same difficulty name, finish, and exit:

https://github.com/user-attachments/assets/ceaf2445-1f74-407b-96e5-46de018e8c09

Reply:
@bdach are you on board with getting this in as-is, based on https://github.com/ppy/osu/issues/33905 also being a thing which this will fix?

Reply:
I'm not sure I agree with removing the TODO given that the [initial reaction](https://github.com/ppy/osu/pull/33558#discussion_r2135176836) to this PR was doubt of the possibility that two beatmaps can actually share the same online ID. Attempting to understand and fix this elsewhere is a multi-day journey in my eyes, I can still go through it but cannot fully commit to it nor do I trust I'll come with positive results.